scripts/installer.sh

```diff
@@ -0,0 +1,251 @@
+#!/bin sh
+# Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+#
+# This script detects the current operating system, and installs
+# Tailscale according to that OS's conventions.
+
+set -eu
+
+# All the code is wrapped in a main function that gets called at the
+# bottom of the file, so that a truncated partial download doesn't end
+# up executing half a script.
+main() {
+	# Step 1: detect the current linux distro, version, and packaging system.
+	#
+	# We rely on /etc/os-release to get OS information. With some
+	# distros, we have to normalize the information we get into the
+	# form that we need to drive our install process.
+
+	OS=
+	VERSION=
+	PACKAGETYPE=
+
+	if [ ! -f /etc/os-release ]; then
+		echo "/etc/os-release not found, cannot get OS information."
+		echo "Please contact us at support@tailscale.com for install help."
+		exit 2
+	fi
+
+	# /etc/os-release populates a number of shell variables. We care about the following:
+	#  - ID: the short name of the OS (e.g. "debian", "freebsd")
+	#  - VERSION_ID: the numeric release version for the OS, if any (e.g. "18.04")
+	#  - VERSION_CODENAME: the codename of the OS release, if any (e.g. "buster")
+	. /etc/os-release
+	case "$ID" in
+		ubuntu)
+			OS="$ID"
+			VERSION="$VERSION_ID"
+			PACKAGETYPE="apt"
+			;;
+		debian)
+			OS="$ID"
+			VERSION="$VERSION_CODENAME"
+			PACKAGETYPE="apt"
+			;;
+		raspbian)
+			OS="$ID"
+			VERSION="$VERSION_CODENAME"
+			PACKAGETYPE="apt"
+			;;
+		centos)
+			OS="$ID"
+			VERSION="$VERSION_ID"
+			PACKAGETYPE="dnf"
+			if [ "$VERSION" = "7" ]; then
+				PACKAGETYPE="yum"
+			fi
+			;;
+		rhel)
+			OS="$ID"
+			VERSION="$(echo $VERSION_ID | cut -f1 -d.)"
+			PACKAGETYPE="dnf"
+			;;
+		fedora)
+			OS="$ID"
+			VERSION=""
+			PACKAGETYPE="dnf"
+			;;
+		amzn)
+			OS="amazon-linux"
+			VERSION="$VERSION_ID"
+			PACKAGETYPE="yum"
+			;;
+		opensuse-leap)
+			OS="opensuse"
+			VERSION="leap/$VERSION_ID"
+			PACKAGETYPE="zypper"
+			;;
+		opensuse-tumbleweed)
+			OS="opensuse"
+			VERSION="tumbleweed"
+			PACKAGETYPE="zypper"
+			;;
+ 		arch)
+			echo "Please install Tailscale from the community repo:"
```

Comment:
For these distros where we know what commands to use… is there a reason we wouldn't want to run the commands for people? I guess because we don't control the repo for the code?

Reply:
Done.